crates/ide-completion/src/context/analysis.rs

```diff
@@ -605,6 +605,26 @@ fn classify_name_ref(
                     },
                     _ => false,
                 };
+
+                let reciever_is_part_of_indivisible_expression = match &receiver {
+                    Some(ast::Expr::IfExpr(_)) => {
+                        let next_sibling = field.dot_token().and_then(|token| {
+                            let dot_token = original_file.covering_element(token.text_range());
+                            let next_sibling = dot_token.as_token().and_then(|t| t.next_token()).and_then(|t| next_non_trivia_sibling(t.into()));
```

Comment:
I've tried to find the next sibling by calling `next_non_trivia_sibling` on `dot_token`, but this return `None`.
```rust
dot_token.as_token().and_then(|t| next_non_trivia_sibling(t.into()))
```
So I find the `next_non_trivia_sibling` on `dot_token.next_token()` instead. I'm not sure if this is 100% correct.

Reply:
There is no need to fetch the dot token from the original file, this logic here can be done fully with the fake syntax tree. The reason why `next_non_trivia_sibling` fails is because the else keyword is wrapped in an error node that is the sibling of the parent of the field expression.

If we use the fake file instead I think it should be enough to just check for the next non trivia token (next from the fake `NameRef` that we get passed in this function. Then we also don't need those next_sibling branches that check for a node that contains `else`, we can just check the `else` token.

Reply:
I updated it, but I found no method to find `next_non_trivia_token`, so I wrote one based on `previous_non_triva_token`.
https://github.com/rust-lang/rust-analyzer/blob/646f97385709fab31db2f10661125c0bbbda6b7c/crates/ide-completion/src/context/analysis.rs#L1304-L1306


I'm not sure if I could reuse `next_non_trivia_sibling` in this case.
https://github.com/rust-lang/rust-analyzer/blob/646f97385709fab31db2f10661125c0bbbda6b7c/crates/ide-completion/src/context/analysis.rs#L1320-L1322


By the way, I think `next_non_trivia_sibling` is duplicated with `syntax::algo::next_non_trivia_sibling`.